packages/flutter/lib/src/gestures/events.dart

```diff
@@ -288,36 +288,32 @@ abstract class PointerEvent {
   final bool synthesized;
 
   @override
-  String toString() => '$runtimeType($position)';
-
-  /// Returns a complete textual description of this event.
-  String toStringFull() {
-    return '$runtimeType('
-             'timeStamp: $timeStamp, '
-             'pointer: $pointer, '
-             'kind: $kind, '
-             'device: $device, '
-             'position: $position, '
-             'delta: $delta, '
-             'buttons: $buttons, '
-             'down: $down, '
-             'obscured: $obscured, '
-             'pressure: $pressure, '
-             'pressureMin: $pressureMin, '
-             'pressureMax: $pressureMax, '
-             'distance: $distance, '
-             'distanceMin: $distanceMin, '
-             'distanceMax: $distanceMax, '
-             'size: $size, '
-             'radiusMajor: $radiusMajor, '
-             'radiusMinor: $radiusMinor, '
-             'radiusMin: $radiusMin, '
-             'radiusMax: $radiusMax, '
-             'orientation: $orientation, '
-             'tilt: $tilt, '
-             'platformData: $platformData, '
-             'synthesized: $synthesized'
-           ')';
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Duration>('timeStamp', timeStamp, defaultValue: Duration.zero));
+    properties.add(IntProperty('pointer', pointer));
+    properties.add(EnumProperty<PointerDeviceKind>('kind', kind));
+    properties.add(IntProperty('device', device, defaultValue: 0));
+    properties.add(DiagnosticsProperty<Offset>('position', position));
+    properties.add(DiagnosticsProperty<Offset>('delta', delta, defaultValue: Offset.zero));
+    properties.add(IntProperty('buttons', buttons, defaultValue: 0));
+    properties.add(DiagnosticsProperty<bool>('down', down));
+    properties.add(FlagProperty('obscured', value: obscured, ifTrue: 'obscured'));
```

Comment:
consider moving all the flag properties together as I think that would read a bit nicer.
e.g. instead of
```
buttons: 2
obscured
delta: Offset(3.0, 5.0)
pressure: 3.0
synthesized
```
a user would see
```
buttons: 2
delta: Offset(3.0, 5.0)
pressure: 3.0
obscured
synthesized
```

Reply:
Done; I was just preserving the order that was there, but reordering makes sense. I also moved `position` (with `delta`) up to the top since `position` is such an important property.